implementations/rust/ockam/ockam_core/src/routing/message/local_message.rs

```diff
@@ -1,116 +1,210 @@
     }
-    /// Return a reference to the underlying transport message.
-    pub fn transport(&self) -> &TransportMessage {
-        &self.transport_message
+
+    /// Return a reference to the message onward route
+    pub fn onward_route_ref(&self) -> &Route {
+        &self.onward_route
+    }
+
+    /// Return the next address on the onward route
+    pub fn next_on_onward_route(&self) -> Result<Address> {
+        Ok(self.onward_route.next()?.clone())
+    }
+
+    /// Return true if an address exists on the onward route
+    pub fn has_next_on_onward_route(&self) -> bool {
+        self.onward_route.next().is_ok()
+    }
+
+    /// Remove the first address of the onward route
+    pub fn pop_front_onward_route(mut self) -> Result<Self> {
+        let _ = self.onward_route.step()?;
+        Ok(self)
     }
-    /// Return a mutable reference to the underlying transport message.
-    pub fn transport_mut(&mut self) -> &mut TransportMessage {
-        &mut self.transport_message
+
+    /// Prepend an address on the onward route
+    pub fn push_front_onward_route(mut self, address: &Address) -> Self {
+        self.onward_route.modify().prepend(address.clone());
+        self
+    }
+
+    /// Replace the first address on the onward route
+    pub fn replace_front_onward_route(self, address: &Address) -> Result<Self> {
+        Ok(self
+            .pop_front_onward_route()?
+            .push_front_onward_route(address))
+    }
+
+    /// Prepend a route to the onward route
+    pub fn prepend_front_onward_route(mut self, route: &Route) -> Self {
+        self.onward_route.modify().prepend_route(route.clone());
+        self
+    }
+
+    /// Set the message onward route
+    pub fn set_onward_route(mut self, route: Route) -> Self {
+        self.onward_route = route;
+        self
     }
-    /// Return a reference to local information added by Workers within the same node.
-    pub fn local_info(&self) -> &[LocalInfo] {
+
+    /// Return the message return route
+    pub fn return_route(&self) -> Route {
+        self.return_route.clone()
+    }
+
+    /// Return a reference to the message return route
+    pub fn return_route_ref(&self) -> &Route {
+        &self.return_route
+    }
+
+    /// Set the message return route
+    pub fn set_return_route(mut self, route: Route) -> Self {
+        self.return_route = route;
+        self
+    }
+
+    /// Prepend an address to the return route
+    pub fn push_front_return_route(mut self, address: &Address) -> Self {
+        self.return_route.modify().prepend(address.clone());
+        self
+    }
+
+    /// Prepend a route to the return route
+    pub fn prepend_front_return_route(mut self, route: &Route) -> Self {
+        self.return_route.modify().prepend_route(route.clone());
+        self
+    }
+
+    /// Remove the first address on the onward route and push another address on the return route
+    pub fn forward(self, address: &Address) -> Result<Self> {
```

Comment:
Yes, I was also uncertain about the name. I called it `step_forward` because `step` makes you think about the fact that we `step` the onward_route and `forward` the fact that we do this to forward the message next.